src/osdp_cp.c

```diff
@@ -844,7 +843,7 @@ static int state_update(struct osdp_pd *pd)
 		}
 		break;
 	case OSDP_CP_STATE_OFFLINE:
-		if (osdp_millis_since(pd->tstamp) > OSDP_CMD_RETRY_WAIT_MS) {
+		if (osdp_millis_since(pd->tstamp) > OSDP_ONLINE_RETRY_WAIT_MS) {
```

Comment:
A retry every second is bit wasteful if the PD is disconnected from the CP (which accounts for 90% of the cases). I will make this into an exponential back-off to achieve best of both worlds.

Reply:
Yes it would usually be wasteful and possibly clutter a log if logging the retries.  The exponential back-off is a good idea.  There is the perhaps very rare but important case such as an installer first connecting two pieces of equipment and wanting to get quick feedback to know when things are working.  A configurable upper bound for the back-off would help with that case for applications with which it is more of a concern.

Reply:
Yes. Start off with one second waits and double till we clap at an upper-bound.